### DISCUSSION TITLE
What  is the  TopLevel.ActualTransparencyLevel   role ?

### DISCUSSION THREAD

**Discussion by Naylor55** [AvaloniaUI/Avalonia#14223] [Q&A]
when I  call it  in   Windows11 , I get   Transparent .  and  in  Windows Server 2016 , I get  None 。So  ,What  happen  when  I call TopLevel.ActualTransparencyLevel?

**Comment by maxkatz6** [comment]
Developers need to expect that not always transparency is supported. The same goes about blur and acrylic.
This property returns an actual transparency level currently set on the window. It might be different from what developer set.
Also, TopLevel.TransparencyLevels accepts multiple ordered values, so ActualTransparencyLevel can be used to know which value was actually set.

**Comment by Naylor55** [comment]
@maxkatz6  yes!!! What are the reasons for the different results on Windows11 and WindowsServer2016 ?  They are using the same set of code。

**Comment by Naylor55** [comment]
Can I conclude that it(WindowsServer2016) does not support `Background Transparency` if it returns None? Is this conclusion reasonable?

**Reply by timunie** [reply]
Yes it is. As said not all OS support it. Even not all Windows versions. I read somewhere in a thread that Windows Server has problems here in WPF as well. Maybe fixed in a newer version

**Comment by Naylor55** [comment]
@timunie ok,Thanks!!!
